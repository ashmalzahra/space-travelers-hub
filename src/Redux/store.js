import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './RocketsSlice';
import missionsSlice from './MissionsSlice';
// import missionsReducer from "./MissionsSlice";

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice.reducer,
  },
});

export default store;
