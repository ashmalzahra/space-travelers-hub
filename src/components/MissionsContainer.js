import React, { useEffect } from "react";
import { Container, Table } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import { fetchMissions } from "../Redux/MissionsSlice";
import Mission from "./Missions";

const MissionsContainer = () => {
    // get mission data from the store
    const missions = useSelector((state) => state.missions.missions);
    const { length } = missions || {};
    const dispatch = useDispatch();
    useEffect(() => {
        if (length === 0) {
            dispatch(fetchMissions());
        }
        
    }, [dispatch]);
    return (
        <Container fluid className="table-responsive-sm">
        <Table className="my-3 table-bordered table-striped">
            <thead>
                <tr>
                    <td><b>Mission</b></td>
                    <td><b>Description</b></td>
                    <td><b>Status</b></td>
                    <td />
                </tr>
            </thead>
            <tbody>
                {missions.map((e) => (
                    <Mission 
                        key={e.mission_id}
                        missionId={e.mission_id}
                        missionName={e.mission_name}
                        description={e.description}
                        reserved={e.reserved}
                    />
                ))}
            </tbody>
        </Table>
        </Container>
    );
};

export default MissionsContainer;