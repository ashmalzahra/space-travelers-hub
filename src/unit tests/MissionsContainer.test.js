import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import MissionsContainer from '../components/MissionsContainer';
import store from '../Redux/store';

it('render', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionsContainer />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders content', () => {
  const component = render(
    <Provider store={store}>
      <MissionsContainer />
    </Provider>,
  );
  expect(component.container).toHaveTextContent('Mission');
  expect(component.container).toHaveTextContent('Description');
  expect(component.container).toHaveTextContent('Status');
});
