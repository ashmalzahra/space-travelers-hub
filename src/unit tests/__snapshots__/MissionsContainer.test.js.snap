// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="table-responsive-sm container-fluid"
>
  <table
    className="my-3 table-bordered table-striped table"
  >
    <thead>
      <tr>
        <td>
          <b>
            Mission
          </b>
        </td>
        <td>
          <b>
            Description
          </b>
        </td>
        <td>
          <b>
            Status
          </b>
        </td>
        <td />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
